eclair-core/src/main/scala/fr/acinq/eclair/channel/fund/InteractiveTxBuilder.scala

```diff
@@ -114,6 +115,31 @@ object InteractiveTxBuilder {
     val minNextFeerate: FeeratePerKw = targetFeerate * 25 / 24
   }
 
+  // @formatter:off
+  sealed trait Purpose {
+    def previousLocalBalance: MilliSatoshi = 0.msat
+    def previousRemoteBalance: MilliSatoshi = 0.msat
+    def commitmentIndex: Long = 0
+    def perCommitmentPoint: PublicKey
+    def commitTxFeerate: FeeratePerKw
+    def common: Common
+  }
```

Comment:
I'm not sure about the name. The real distinction is whether:
- we create the first commitment
- we replace the first commitment
- we spend the existing commitment (splice)

We can make it conceptually independent from the notion of splice. Maybe I'm going too deep.

Reply:
It feels like you're trying hard to create an abstraction, but there is a 1:1 mapping between the abstraction and the concrete use-case, so it feels like the abstraction is just making things less clear...What about being more explicit (at least initially) about the scenarios we support?

- `Purpose` can probably stay `Purpose`
- `InitialCommitment` -> `FundingTx`
- `RbfInitialCommitment` -> `FundingTxRbf`
- (future) `ExistingCommitment` -> `SpliceTx`
- (future) `RbfExistingCommitment` -> `SpliceTxRbf`

Those are the only use-cases we can think of right now, and they all need different parameters. We will probably later use `InteractiveTx` for mutual close, but this will also require a slightly different handling, so it will use a different `Purpose`.

Reply:
Went with your suggestion in d47aec354.